TeslaCommand.go

```diff
@@ -1,179 +1,209 @@
+
+	// Loop every N seconds.
+	logger.Printf(waitmessage)
+	for _ = range time.Tick(time.Duration(checkInterval) * time.Second) {
+	
+	//LSE. Add vehicle wakeup call
+	_, err = vehicle.Wakeup()
+	if err != nil {
+		logger.Println(err)
+		os.Exit(1)
+	}
+	time.Sleep(60 * time.Second) //Give it a chance to wake up   LSE
+	// LSE
+	
+		logger.Printf("Checking vehicle %v location after waiting %v seconds.\n", vehicleState.VehicleName, checkInterval)
+
+		driveState, err := vehicle.DriveState()
+		if err != nil {
+			logger.Println(err)
+			logger.Printf(waitmessage)
+			continue
+		}
+
+		distance := HaversinFormula.Distance(geoFenceLatitude, geoFenceLongitude, driveState.Latitude, driveState.Longitude)
+
+		logger.Printf("Distance: %v\n\n", distance)
+
+		// If the distance is outside the radius, that means vehicle is outside the GeoFence.  Ok to get out
+		if distance > float64(radius) {
+			// LSEingeofenceandstopped = false
+			logger.Printf(waitmessage)
+			continue
+		}
+
+        // The following code was removed in order to make loop do the whole test each time, including sending email if disconnected LSE
+		// This is to prevent the below logic, if it's already executed, no need to keep doing it
+		// LSEif ingeofenceandstopped == true {
+			// LSElogger.Printf(waitmessage)
+			// LSEcontinue
+		// LSE}
+
+		// Check if the vehicle is stopped.
+		if (driveState.ShiftState == nil || driveState.ShiftState == "P") && driveState.Speed == 0 {
+			// LSEingeofenceandstopped = true
+
+			// In the GeoFence and stopped.  Get the vehicle charge state.
+			chargeState, err := vehicle.ChargeState()
+			if err != nil {
+				logger.Println(err)
+				logger.Printf(waitmessage)
+				continue
+			}
+
+			// Check if the vehicle is plugged in.
+			logger.Printf("Vehicle %v is within %v meters of GeoFence with a battery level of %v percent and charging state of %v.", vehicleState.VehicleName, int(distance), chargeState.BatteryLevel, chargeState.ChargingState)
+			if chargeState.ChargingState != "Disconnected" {
+				logger.Printf("Charge state is %v. Exit", chargeState.ChargingState)
+				os.Exit(1)
+			}
+				// Disconnected, stopped, and within the radius - send alert
+				if chargeState.BatteryLevel >= alertThreshold {
+					logger.Printf("Battery level is above %v alert threshold. Not sending alert.", chargeState.BatteryLevel)
+					logger.Printf(waitmessage)
+					continue
+				}
+
+				subject := "Tesla Command - " + vehicleState.VehicleName
+				body := fmt.Sprintf("Vehicle %v is within %v meters of GeoFence with a battery level of %v percent.  Please plug in.", vehicleState.VehicleName, int(distance), chargeState.BatteryLevel)
+				logger.Println(body)
+
+				// Send mail
+				err = NotifyLib.SendMail(logger, mailServer, mailServerPort, mailServerLogin, mailServerPassword, fromAddress, toAddress1, toAddress2, subject, body)
+				if err != nil {
+					logger.Println(err)
+				}
+
+//LSE				// Send text

```

Comment:
Remove

Reply:
With this code in place, the program writes an error message unless the user has a twilio account. I don't, so the error message (at least) has to be skipped.